StoreListings.CLI/Commands.cs

```diff
@@ -6,31 +6,246 @@ namespace StoreListings.CLI;
 
 public class Commands
 {
+    /// <summary>
+    /// Query products using keyword from Microsoft Store.
+    /// </summary>
+    /// <param name="query">The keyword to query.</param>
+    /// <param name="deviceFamily">-d, The device family.</param>
+    /// <param name="market">-m, The store market/region to query from.</param>
+    /// <param name="language">-l, The language, for listings that use localization.</param>
+    public async Task Query(
+        [Argument] string query,
+        CancellationToken cancellationToken,
+        DeviceFamily deviceFamily = DeviceFamily.Desktop,
+        Market market = Market.US,
+        Lang language = Lang.en
+    )
+    {
+        WriteLoadingProgressBar();
+        Result<StoreEdgeFDQuery> result = await StoreEdgeFDQuery.GetSearchProduct(
+            query,
+            deviceFamily,
+            market,
+            language,
+            cancellationToken
+        );
+        HideProgressBar();
+        if (result.IsSuccess)
+        {
+            List<Card> cards = result.Value.Cards;
+            foreach (var card in cards)
+            {
+                WriteField("Product ID", card.ProductId);
+                WriteField("Title", card.Title);
+                if (card.DisplayPrice != null)
+                {
+                    WriteField("Display price", card.DisplayPrice);
+                }
+                if (card.AverageRating != null)
+                {
+                    WriteField("Average rating", card.AverageRating.ToString());
+                }
+                WriteField("Image", card.Image.Url);
+            }
+        }
+        else
+        {
+            Console.WriteLine(result.Exception);
+        }
+    }
+
+    /// <summary>
+    /// Query suggestion for a keyword from Microsoft Store.
```

Comment:
I don't think this description is correct.

Other than that, I think this is pretty much good to go.